storage/storage.go

```diff
@@ -93,7 +93,7 @@ type PeerStore interface {
 	//   leechers
 	//
 	// Returns ErrResourceDoesNotExist if the provided InfoHash is not tracked.
-	AnnouncePeers(infoHash bittorrent.InfoHash, seeder bool, numWant int, p bittorrent.Peer) (peers []bittorrent.Peer, err error)
+	AnnouncePeers(infoHash bittorrent.InfoHash, seeder bool, numWant int, p bittorrent.Peer, peers []bittorrent.Peer) ([]bittorrent.Peer, error)
```

Comment:
This could use an update to the godoc stating what happens for error cases. You opted to return the (possibly already modified) peer slice together with the error. I think that's fine, but it needs to be documented. We could alternatively also document that only `nil, <some error>` or `<some peer slice>, nil` may be returned. In that case, we'd need to change the implementations.
The idiomatic Go approach would be to return `nil, <some error>` on error. I'd be fine with your approach, though.

Reply:
Good point. I didn't know that. Changed back to `nil, peers` as that sounds more idiomatic and added a line explaining the new optional parameter.